### PR TITLE
[core] Fix FieldCountAgg null value init process

### PR DESCRIPTION
**Pull request by xuzifu666**
Currently we had FieldAggregator to compute all field cout，but not support filter null field count，this pr is aim to support it.

Linked issue: https://github.com/apache/paimon/issues/3537

Null value is filter count by agg operation

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/mergetree/compact/aggregate/FieldNonNullCountAgg.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            }
+        }
+        return count;
+    }
+
+    @Override
+    public Object retract(Object accumulator, Object inputField) {
+        Object count;
+        if (accumulator == null || inputField == null) {
+            count = (accumulator == null ? (inputField == null ? 0 : 1) : accumulator);
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/mergetree/compact/aggregate/FieldNonNullCountAgg.java:62]: should be -1?

Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/compact/aggregate/FieldNonNullCountAgg.java:62]: Yes, it should be -1，thanks for the reminder, it alse error in FieldCountAgg，would fix it all. @JingsongLi

**Review by JingsongLi — COMMENTED** [review]
Hi I think there is bug in `FieldCountAgg`, the count should 0 when accumulator is null and inputField is null.

**Comment by JingsongLi** [comment]
actually `FieldCountAgg` is the `FieldNonNullCountAgg`.

**Comment by xuzifu666** [comment]
> actually `FieldCountAgg` is the `FieldNonNullCountAgg`.

Add the logic to the FieldCountAgg and remove FieldNonNullCountAgg.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**